apis/kms/v1alpha1/cryptokey_types.go

```diff
@@ -0,0 +1,465 @@
+	// +immutable
+	KeyRingRef *xpv1.Reference `json:"keyRingRef,omitempty"`
+
+	// KeyRingSelector selects a reference to a KeyRing
+	// +optional
+	// +immutable
+	KeyRingSelector *xpv1.Selector `json:"keyRingSelector,omitempty"`
+
+	// Labels: Labels with user-defined metadata. For more information,
+	// see
+	// [Labeling Keys](/kms/docs/labeling-keys).
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Purpose: Immutable. The immutable purpose of this CryptoKey.
+	//
+	// Possible values:
+	//   "CRYPTO_KEY_PURPOSE_UNSPECIFIED" - Not specified.
+	//   "ENCRYPT_DECRYPT" - CryptoKeys with this purpose may be used
+	// with
+	// Encrypt and
+	// Decrypt.
+	//   "ASYMMETRIC_SIGN" - CryptoKeys with this purpose may be used
+	// with
+	// AsymmetricSign and
+	// GetPublicKey.
+	//   "ASYMMETRIC_DECRYPT" - CryptoKeys with this purpose may be used
+	// with
+	// AsymmetricDecrypt and
+	// GetPublicKey.
+	// +immutable
+	Purpose string `json:"purpose"`
+
+	// RotationPeriod: next_rotation_time will be advanced by this period
+	// when the service
+	// automatically rotates a key. Must be at least 24 hours and at
+	// most
+	// 876,000 hours.
+	//
+	// If rotation_period is set, next_rotation_time must also be set.
+	//
+	// Keys with purpose
+	// ENCRYPT_DECRYPT support
+	// automatic rotation. For other keys, this field must be omitted.
+	// +optional
+	RotationPeriod *string `json:"rotationPeriod,omitempty"`
+
+	// NextRotationTime: At next_rotation_time, the Key Management Service
+	// will automatically:
+	//
+	// 1. Create a new version of this CryptoKey.
+	// 2. Mark the new version as primary.
+	//
+	// Key rotations performed manually via
+	// CreateCryptoKeyVersion and
+	// UpdateCryptoKeyPrimaryVersion
+	// do not affect next_rotation_time.
+	//
+	// Keys with purpose
+	// ENCRYPT_DECRYPT support
+	// automatic rotation. For other keys, this field must be omitted.
+	// +optional
+	NextRotationTime *string `json:"nextRotationTime,omitempty"`
+
+	// VersionTemplate: A template describing settings for new
+	// CryptoKeyVersion instances.
+	// The properties of new CryptoKeyVersion instances created by
+	// either
+	// CreateCryptoKeyVersion or
+	// auto-rotation are controlled by this template.
+	// +optional
+	VersionTemplate *CryptoKeyVersionTemplate `json:"versionTemplate,omitempty"`
+}
+
+// CryptoKeyObservation is used to show the observed state of the
+// CryptoKey resource on GCP. All fields in this structure should only
+// be populated from GCP responses; any changes made to the k8s resource outside
+// of the crossplane gcp controller will be ignored and overwritten.
+type CryptoKeyObservation struct {
+	// CreateTime: Output only. The time at which this CryptoKey was
+	// created.
+	CreateTime string `json:"createTime,omitempty"`
+
+	// Labels: Labels with user-defined metadata. For more information,
+	// see
+	// [Labeling Keys](/kms/docs/labeling-keys).
+	Labels map[string]string `json:"labels,omitempty"`
```

Comment:
Do you think we need this in status if also in spec?

Reply:
Hmm, I don't think so, removed.